routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
ok- so a few thoughts:

when a project is created, i assume that a user will be redirected to a view that has a file open and ready to edit (likely `index.html`). this default file should be created at the time the project is created. that way a foreign key will exist. all projects, to be projects need at least one file, IMO.

thoughts?